cypress/e2e/requests.cy.js

```diff
@@ -2,50 +2,102 @@ describe('Viewing all requests', () => {
   describe('as a logged out user', () => {
     it('should show an error message.', () => {
       // Visit a protected route in order to allow cypress to set the cookie and mock the login
-      cy.visit("/requests")
+      cy.visit('/requests')
       cy.get('div.alert-heading').contains('Unauthorized').then(() => {
-        cy.log("A logged out user is not able to view requests.")
+        cy.log('A logged out user is not able to view requests.')
       })
     })
   })
   
   describe('as a logged in user', () => {
     let scientistApiBaseURL = `https://${Cypress.env('NEXT_PUBLIC_PROVIDER_NAME')}.scientist.com/api/v2`
+    // declare variables that can be used to change how the response is intercepted.
     let requestList
+    let loading
+    let error
 
     beforeEach(() => {
       // Call the custom cypress command to log in
       cy.login(Cypress.env('TEST_SCIENTIST_USER'), Cypress.env('TEST_SCIENTIST_PW'))
-
       // Intercept the response from the endpoint to view all requests
+      // TODO(summer-cook): extract out this base url into the config to use as an environment variable. it was not cooperating before
       cy.intercept('GET', `${scientistApiBaseURL}/quote_groups/mine.json`, (req) => {
-        if (requestList === true) {
-          req.reply({fixture: 'all-requests/requests.json'})
-        } else {
-          req.reply({fixture: 'all-requests/no-requests.json'})
+        if ((requestList === undefined) && (loading === true)) {
+          // reply with an empty response: both data and error will be undefined.
+          req.reply()
+        } else if ((requestList === undefined) && (loading === undefined) && (error === true)) {
```

Comment:
why does loading equal undefined instead of false?

Reply:
all these variables are doing are switching between the different intercept cases, so it could be either undefined or false, it wouldn't really matter as long as its not true. But, it does make it easier to read to be false so i just changed it

Reply:
ok.
yes, the test passes either way but I was thinking from the POV of someone who is unfamiliar with this code. just like we check the api specs to see how things work, someone may check these specs. so it's better, imo, to keep the tests in line with what the actual pages are demonstrating.